File: api/models/comment.go
package models

import (
	"time"
)

type Comment struct {
	ID        uint      `json:"id" gorm:"primaryKey;autoIncrement"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	Content string    `json:"content"`
	Post    Post      `json:"post" gorm:"foreignKey:PostID"`
	User    User      `json:"user" gorm:"foreignKey:UserID"`
	Comment []Comment `json:"comments" gorm:"foreignKey:CommentID"`

	PostID uint `json:"post_id"`
	UserID uint `json:"user_id"`
}

File: frontend/app/models/comment.ts
import { Post } from "./post";
import { User } from "./user";

export interface Comment {
    id: number;
    created_at: string;
    updated_at: string;
    content: string;
    post: Post;
    user: User;
    post_id: number;
    user_id: number;
}